Tests/FluentAssertions.Specs/FindAssembly.cs

```diff
@@ -1,11 +1,50 @@
-﻿using System.Reflection;
+﻿using System;
+using System.Reflection;
 
 namespace FluentAssertions.Specs;
 
 public static class FindAssembly
 {
-    public static Assembly Containing<T>()
+    public static Assembly Containing<T>() => typeof(T).Assembly;
+
+    public static Assembly Stub(string publicKey = "") => new AssemblyStub(publicKey);
+
+    private sealed class AssemblyStub : Assembly
     {
-        return typeof(T).Assembly;
+        private readonly AssemblyName assemblyName = new();
+
+        public override string FullName => nameof(AssemblyStub);
+
+        public AssemblyStub(string publicKey)
+        {
+            assemblyName.SetPublicKey(FromHexString(publicKey));
+        }
+
+        public override AssemblyName GetName() => assemblyName;
+
+#if NET6_0_OR_GREATER
+        private static byte[] FromHexString(string chars)
+            => string.IsNullOrEmpty(chars)
+            ? Array.Empty<byte>()
+            : Convert.FromHexString(chars);
+#else
+        private static byte[] FromHexString(string chars)
+        {
+            if (chars.Length % 2 != 0)
+            {
+                throw new ArgumentException("Input string length must be even.");
+            }
```

Comment:
Should we add this (or a similar) check as guard inside the production code?

Reply:
I don't think it's necessary. 
If one provides an expected public key of odd length the assertion will fail and include both subject and expected key in the failure message